pkgs/development/libraries/fmt/default.nix

```diff
@@ -17,6 +18,15 @@ stdenv.mkDerivation rec {
              else "";
   cmakeFlags = [ "-DFMT_TEST=yes"
                  "-DBUILD_SHARED_LIBS=${if enableShared then "ON" else "OFF"}" ];
+  postInstall = ''
+    mkdir -pv $out/lib/pkgconfig
+    cd $src
+    cd /build/source
+    pc="fmt.pc"
+    substituteInPlace "$pc" --replace "@out@" "$out"
+    substituteInPlace "$pc" --replace "@version@" "${version}"
```

Comment:
You can use `substituteAll` here which does the '@variable@' substitution out of the box.